models/ref/python.mdx

```diff
@@ -33,4 +35,41 @@ project = "my-awesome-project"
 
 with wandb.init(entity=entity, project=project) as run:
    run.log({"accuracy": 0.9, "loss": 0.1})
-````
+```
+
+## Install optional dependencies
+
+Install optional dependencies that extend the functionality of the W&B Python SDK. Use the following command to install an optional dependency. Replace `extra` with the desired [optional dependency](#supported-optional-dependencies):
+
+```shell
+pip install wandb[extra]
+```
+
+For example, to install W&B with Google Cloud Storage support, run:
+
+```shell
+pip install wandb[gcp]
+```
+
+Install more than one optional dependency by separating them with commas:
+
+```shell
+pip install wandb[gcp,aws,media]
+```
+
+### Supported optional dependencies
+
+The following table lists supported dependencies for the W&B Python SDK:
+
+| Package | Purpose | Packages included |
+|---------|---------| ------------------|
+| `kubeflow` | Integrations for Kubeflow and object storage access. | [kubernetes](https://pypi.org/project/kubernetes/), [minio](https://pypi.org/project/minio/), [google-cloud-storage](https://pypi.org/project/google-cloud-storage/), [sh](https://pypi.org/project/sh/) |
+| `gcp` |  Google Cloud Storage support. | [google-cloud-storage](https://pypi.org/project/google-cloud-storage/) |
+| `aws` | AWS S3 and other AWS service support. | [boto3](https://pypi.org/project/boto3/), [botocore](https://pypi.org/project/botocore/) |
+| `azure` | Azure Blob Storage support. | [azure-identity](https://pypi.org/project/azure-identity/), [azure-storage-blob](https://pypi.org/project/azure-storage-blob/) |
+| `media` | Packages for media handling, video/image processing, plotting, and cheminformatics (RDKit).| [numpy](https://pypi.org/project/numpy/), [moviepy](https://pypi.org/project/moviepy/), [imageio](https://pypi.org/project/ImageIO/), [pillow](https://pypi.org/project/pillow/), [bokeh](https://pypi.org/project/bokeh/), [soundfile](https://pypi.org/project/soundfile/), [plotly](https://pypi.org/project/plotly/), [rdkit](https://pypi.org/project/rdkit/) |
+| `sweeps`| Manage sweeps. | [sweeps](https://pypi.org/project/sweeps/) | 
+| `perf` | High-performance JSON serializer. | [orjson](https://pypi.org/project/orjson/) |
+| `workspaces` | W&B Reports and Workspaces SDK. | [wandb-workspaces](https://pypi.org/project/wandb-workspaces/) |
+
+See `optional-dependencies` in the [W&B Python SDK GitHub repository](https://github.com/wandb/wandb/blob/main/pyproject.toml#L61) for an up-to-date list of optional dependencies.
```

Comment:
As discussed, this list should be auto-generated by script, removing the need for this sentence. (In general, pointing people to github source files for "documentation" probably isn't what we want? After all, consider all the exceptions to that list in the source Dima is pointing out above ^ )

Reply:
@johndmulhausen QQ: A handful of these are being manually removed (i.e. no longer 1:1 between `.mdx` file and `pyproject.toml`) because they are in various states of "working". Given that, is it worth generating a script?